var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -72,20 +72,7 @@ class Dealii(CMakePackage):
     # https://github.com/dealii/dealii/issues/1591
     # Require at least 1.59
     # +python won't affect @:8.4.2
-    depends_on("boost@1.59.0:+thread+system+serialization+iostreams",
-               when='@:8.4.2~mpi')
-    depends_on("boost@1.59.0:+thread+system+serialization+iostreams+mpi",
-               when='@:8.4.2+mpi')
-    # since @8.5.0: (and @develop) python bindings are introduced:
-    depends_on("boost@1.59.0:+thread+system+serialization+iostreams",
-               when='@8.5.0:~mpi~python')
-    depends_on("boost@1.59.0:+thread+system+serialization+iostreams+mpi",
-               when='@8.5.0:+mpi~python')
-    depends_on("boost@1.59.0:+thread+system+serialization+iostreams+python",
-               when='@8.5.0:~mpi+python')
-    depends_on(
-        "boost@1.59.0:+thread+system+serialization+iostreams+mpi+python",
-        when='@8.5.0:+mpi+python')
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams")
```

Comment:
Frankly, i don't understand why you do this. You say

> Well, first of all, it doesn't forward the variants properly

but in my experience things work perfectly fine in this regards.

I agree that if you would try to specify which `boost` is needed here, you would need to do `spack spec dealii+mpi+python ^boost@1.60.0` or alike, it's PITA but if you put your preferences to `packages.yaml`, then all works fine, at least from my experience.